altair/examples/line_chart_with_datum.py

```diff
@@ -0,0 +1,31 @@
+"""
+Line Chart with XDatum and YDatum
+---------------------------------
+An example of using ``XDatum`` and ``YDatum`` to highlight certain values, including a ``DateTime`` value.
+This is adapted from two corresponding Vega-Lite Examples:
+`Highlight a Specific Value <https://vega.github.io/vega-lite/docs/datum.html#highlight-a-specific-data-value>`_.
+"""
+# category: line charts
+
+import altair as alt
+from vega_datasets import data
+
+source = data.stocks()
+no_source = alt.InlineData([{}])
+
+lines = alt.Chart(source).mark_line().encode(
+    x = alt.X("date"),
+    y = alt.Y("price"),
+    color = "symbol"
+)
+
+xrule = alt.Chart(no_source).mark_rule(color='cyan', strokeWidth=2).encode(
+    x=alt.XDatum(alt.DateTime(year=2006, month = "November"))
+)
+
+yrule = alt.Chart(no_source).mark_rule(strokeDash=[12,6], size=2).encode(
+    y = alt.YDatum(350)
```

Comment:
Altair currently has an `alt.value(x)` that can be used in place of the `alt.YValue()` classes. I wonder if it would make sense to have `alt.datum(x)` for use in place of `alt.YDatum(x)`?

It's maybe a bit strange, because `alt.datum` already exists for a different purpose, but we could do this by adding a `__call__` method to [`DatumType`](https://github.com/altair-viz/altair/blob/159831869f84298af19b6d231c35204cf1c7b40d/altair/expr/core.py#L4) with an implementation similar to this: https://github.com/altair-viz/altair/blob/8a8642b2e7eeee3b914850a8f7aacd53335302d9/altair/vegalite/v4/api.py#L209-L211

What do you think?

Reply:
I find myself using `alt.value` very often, like for example if I just want to set an explicit color (not sure if that's good style or not), so having an `alt.datum` seems nice.  If the change breaks a lot of existing code then I guess it's not worth it here.  (I admit I don't really see the big picture of why both Datum and Value exist on the Vega-Lite side.  I think I haven't digested the essential differences between the concepts.)

Reply:
Now that I understand this option (and that it doesn't break the existing `alt.datum`) I really like it.  Do you think any/all of the three datum examples should be changed to use `alt.datum` instead of the longer syntax (like `alt.Theta2Datum`)?